pkg/sql/grant_revoke_system.go

```diff
@@ -247,25 +247,28 @@ func (p *planner) SynthesizePrivilegeDescriptor(
 	privilegeObjectPath string,
 	privilegeObjectType privilege.ObjectType,
 ) (*catpb.PrivilegeDescriptor, error) {
+	_, desc, err := p.Descriptors().GetImmutableTableByName(ctx, p.Txn(),
+		syntheticprivilege.SystemPrivilegesTableName, tree.ObjectLookupFlagsWithRequired())
+	if err != nil {
+		return nil, err
+	}
+	if !desc.IsUncommittedVersion() {
```

Comment:
Can you help me understand this code? If it is **not** an uncommitted version, then we'll go to this method which is going to synchronize with concurrent cache updates but we'll not write that to the cache. Then if it **is** an uncommitted version (which means that there are changes, right?), then we will check the cache and maybe write it back?